scripts/profile-compiler-build.sh

```diff
@@ -37,7 +39,14 @@ build_compiler() {
   make install
 }
 
+temp_dir=$(mktemp -d -p $root/.. -t tmp.XXXXXXX)
```

Comment:
Not sure about the `-p $root/..` parameter;
is relying on `TMPDIR` a problem in practice?

Reply:
As discussed offline, `TMPDIR` is not always set and may have space issues. So, the code has been changed to make the temp directory in `TMPDIR` if it is set and as a sibling of the repository directory if it is not set.